kapitan/remoteinventory/fetch.py

```diff
@@ -0,0 +1,201 @@
+
+        :returns: None
+    """
+    source, inventories = inv_mapping
+    fetch_git_source(source, save_dir)
+    for inv in inventories:
+        repo_path = os.path.join(save_dir, os.path.basename(source))
+        repo = Repo(repo_path)
+        output_path = os.path.join(inventory_path, inv["output_path"])
+        copy_src_path = repo_path
+
+        if "ref" in inv:
+            ref = inv["ref"]
+            repo.git.checkout(ref)
+        else:
+            repo.git.checkout("master")
+
+        if "subdir" in inv:
+            sub_dir = inv["subdir"]
+            full_subdir = os.path.join(repo_path, sub_dir)
+            if os.path.isdir(full_subdir):
+                copy_src_path = full_subdir
+            else:
+                raise GitSubdirNotFoundError(
+                    "Inventory {} : subdir {} not found in repo".format(source, sub_dir)
+                )
+        # Recursively copy the fetched inventory files (src) to the output path (dst)
+        # if those items doen't exist or if it's older than src
+        copy_tree(copy_src_path, output_path, update=True)
+        logger.info("Inventory {} : saved to {}".format(source, output_path))
+
+
+def fetch_git_source(source, save_dir):
+    """Clones a git repo at source and stores it at save_dir"""
+    logger.info("Inventory {} : fetching now".format(source))
+    try:
+        Repo.clone_from(source, os.path.join(save_dir, os.path.basename(source)))
+        logger.info("Inventory {} : successfully fetched".format(source))
+    except GitCommandError as e:
+        logger.error(e)
+        raise GitFetchingError("Inventory {} : fetching unsuccessful{}".format(source, e.stderr))
+
+
+def fetch_http_inventories(inv_mapping, inventory_path, save_dir):
+    """Fetches inventory items over http[s] and saves it to save_dir which is later recursively
+       copied to the output_path stored in inv_mapping
+
+       :param inv_mapping: tupple containing source and list of invetory variables
+       :param inventory_path: default or user defined inventory path
+       :param save_dir: directory to same the fetched inventory items into
+
+       :returns: None
+    """
+    source, invs = inv_mapping
+    content_type = fetch_http_source(source, save_dir)
+    path_hash = hashlib.sha256(os.path.dirname(source).encode()).hexdigest()[:8]
+    copy_src_path = os.path.join(save_dir, path_hash + os.path.basename(source))
+    for inv in invs:
+        output_path = os.path.join(inventory_path, inv["output_path"])
+        if inv.get("unpack", False):
+            # ensure that the directory we are extracting to exists
+            os.makedirs(output_path, exist_ok=True)
+            is_unpacked = unpack_downloaded_file(copy_src_path, output_path, content_type)
+            if is_unpacked:
+                logger.info("Inventory {} : extracted to {}".format(source, output_path))
+            else:
+                logger.info(
```

Comment:
I'd change this to logger.warning. Since it's most likely user error that they're trying to fetch an unsupported file and we want to make it clear this failed here.